codegen/src/types/derives.rs

```diff
@@ -16,35 +16,55 @@ pub struct DerivesRegistry {
     specific_type_derives: HashMap<syn::TypePath, Derives>,
 }
 
+impl Default for DerivesRegistry {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 impl DerivesRegistry {
-    /// Creates a new `DerivesRegistry` with the supplied `crate_path`.
+    /// Creates a new `DerivesRegistry` with no default derives.
+    pub fn new() -> Self {
+        Self {
+            default_derives: Derives::new(),
+            specific_type_derives: Default::default(),
+        }
+    }
+
+    /// Creates a new `DerivesRegistry` with default derives.
     ///
     /// The `crate_path` denotes the `subxt` crate access path in the
     /// generated code.
-    pub fn new(crate_path: &CratePath) -> Self {
+    pub fn with_default_derives(crate_path: &CratePath) -> Self {
         Self {
-            default_derives: Derives::new(crate_path),
+            default_derives: Derives::with_defaults(crate_path),
             specific_type_derives: Default::default(),
         }
     }
 
     /// Insert derives to be applied to all generated types.
-    pub fn extend_for_all(&mut self, derives: impl IntoIterator<Item = syn::Path>) {
-        self.default_derives.derives.extend(derives)
+    pub fn extend_for_all(
+        &mut self,
+        derives: impl IntoIterator<Item = syn::Path>,
+        attributes: impl IntoIterator<Item = syn::Attribute>,
+    ) {
+        self.default_derives.derives.extend(derives);
+        self.default_derives.attributes.extend(attributes);
     }
 
     /// Insert derives to be applied to a specific generated type.
     pub fn extend_for_type(
         &mut self,
         ty: syn::TypePath,
         derives: impl IntoIterator<Item = syn::Path>,
-        crate_path: &CratePath,
+        attributes: impl IntoIterator<Item = syn::Attribute>,
     ) {
         let type_derives = self
             .specific_type_derives
             .entry(ty)
-            .or_insert_with(|| Derives::new(crate_path));
-        type_derives.derives.extend(derives)
+            .or_insert_with(Derives::new);
```

Comment:
Ah nice catch; previously this was starting with default derives, but that looks to be unnecessary, so I like this simplificaiton!